### ISSUE TITLE
How do I trigger a full-page reload under webpack-serve from an integrated plugin when the html-webpack-plugin HTML output should change?

### ISSUE THREAD

**Issue by ezzatron** [jantimon/html-webpack-plugin#990]
Firstly, sorry for not using the issue template, and for the terrible issue title. This is neither a bug report nor a feature request. It's more of a question about integration with `html-webpack-plugin`.

I've written a Webpack plugin that integrates with `html-webpack-plugin` ([@eloquent/git-version-webpack-plugin](https://github.com/eloquent/git-version-webpack-plugin)). Basically it reads what Git tag/branch/commit you're currently on, and adds a `VERSION` property to the `window` object when used in conjunction with `html-webpack-plugin`.

In my plugin, I'm adding some entries to the `fileDependencies` and `contextDependencies` of the Webpack compilation. When running `webpack-serve`, this causes a build to be triggered when switching between Git tags/branches/commits.

This works pretty well for the most part, but when using hot module replacement, `webpack-serve` (or the `webpack-dev-middleware` I guess) determines that no modules have updated, and does not trigger a reload of the page.

Do you know how I can trigger a reload of the `html-webpack-plugin` HTML when these dependencies change? I assume that I need to access the child compilation used by `html-webpack-plugin`, but even after digging through a bunch of plugins that integrate with `html-webpack-plugin`, I cannot figure out how to do so.

**Comment by jantimon** [comment]
I am not sure how the html-webpack-plugin would be able to trigger a reload.

**Comment by jantimon** [comment]
You could try the following (however I haven't tried it by myself)
In webpack there is the `compilation` hook and the `this-compilation` hook.
The compilation hook is fired also for child compilers like the html-webpack-plugin hook.

**Comment by ezzatron** [comment]
Thanks for the advice. I did figure out how to access the child compilation, but unfortunately it didn't lead me to any solutions.

I also found a bunch of the existing GitHub issues surrounding HMR + `html-webpack-plugin` (#100, #638, #785, webpack-contrib/webpack-hot-middleware#237, webpack-contrib/webpack-hot-middleware#249) which all seem vaguely related. Unfortunately I couldn't find any viable workarounds in any of that discussion either.

I think this issue might actually be a duplicate of other existing issues surrounding the fact that when using something like `webpack-dev-server` or `webpack-serve`, the browser window isn't reloaded when `html-webpack-plugin`'s output changes. Because if I reload the window manually, the HTML updates as expected.

I guess in the past the solution was to use [reload-html-webpack-plugin](https://github.com/andrewcoelho/reload-html-webpack-plugin), but it seems essentially abandoned these days, and its approach was pretty hacky in the first place.
